### DISCUSSION TITLE
Ace editor with AI, looking if i can use marked for AI chat string parse

### DISCUSSION THREAD

**Discussion by ldijkman** [markedjs/marked#3095] [Ideas]
Ace editor with AI, looking if i can use marked for AI chat string parse

ai chat stream should be parsed for tripple backtick code fields
i want to paste at ace  cursor position in future

https://ldijkman.github.io/Ace_Seventh_Heaven/The_AfterWorld.html

 
![Screenshot from 2023-11-19 14-04-09](https://github.com/markedjs/marked/assets/45427770/397034f5-54ff-4549-9e7b-f4ffc76a53af)

![Screenshot from 2023-11-13 18-39-01](https://github.com/markedjs/marked/assets/45427770/cab55d4d-6bff-4d0d-a89e-b9f35a31cea6)

**Comment by UziTech** [comment]
I'm not exactly sure what you are asking. Marked can parse any markdown and produces HTML.

**Comment by ldijkman** [comment]
A.I. chat response now parsed by your js code
when ai chat stream is finnished

question to ai is
javascript for loop examples

![Screenshot from 2023-11-20 19-32-27](https://github.com/markedjs/marked/assets/45427770/83d6edd1-5b03-4983-a5f0-9c98e0b37b27)

**Comment by ldijkman** [comment]
have a bit of a problem
when the parse string contains html tags without leading and ending tripple backticks
then the html code is treaded as html

this goes ok
![Screenshot from 2023-11-21 17-36-35](https://github.com/markedjs/marked/assets/45427770/7b297d70-b99e-4526-9c3f-14f601c4b23d)

**Reply by UziTech** [reply]
That is expected. HTML is valid markdown. [spec](https://spec.commonmark.org/0.30/#html-blocks)
